packages/iobridge-commons/src/message/cross-chain-message.ts

```diff
@@ -0,0 +1,56 @@
+import { recoverPersonalSignature } from '@metamask/eth-sig-util';
+import { helpers } from '@ckb-lumos/lumos';
+
+interface EthereumRpc {
+  (payload: { method: 'personal_sign'; params: [string /* from */, string /* message */] }): Promise<string>;
+}
+
+export interface EthereumProvider {
+  selectedAddress: string;
+  request: EthereumRpc;
+}
+
+export type CrossChainMessage = { txHash: string; toAddress: string };
+
+export const TxHashTypeError = new TypeError('txHash type is not string');
+export const ToAddressTypeError = new TypeError('toAddress type is not string');
+
+export function validateCrossChainMessage(message: string): void {
+  const messageObj = JSON.parse(message);
+
+  if (typeof messageObj.txHash !== 'string') {
+    throw TxHashTypeError;
+  }
+
+  if (typeof messageObj.toAddress !== 'string') {
+    throw ToAddressTypeError;
+  }
+}
+
+export function verifyCrossChainMessage(payload: {
+  expectedAddress: string; // eth address
+  signature: string;
+  message: string;
+}): boolean {
+  try {
+    const msg = `0x${Buffer.from(payload.message, 'utf8').toString('hex')}`;
+    const recoveredAddr = recoverPersonalSignature({
+      data: msg,
+      signature: payload.signature,
+    });
+
+    return payload.expectedAddress === recoveredAddr;
+  } catch (_) {
```

Comment:
```suggestion
  } catch {
```

NodeJS 12 is enabled to handle the `catch` syntax, we can add `engines: {nodejs: ">12.0.0"}` the avoid `catch(_)` syntax